bookkeeper-server/src/main/java/org/apache/bookkeeper/util/ZkUtils.java

```diff
@@ -240,7 +241,12 @@ public void operationComplete(int rc, List<String> ledgers) {
 
         synchronized (ctx) {
             while (!ctx.done) {
-                ctx.wait();
+                try {
+                    ctx.wait(timeOutSec > 0 ? TimeUnit.SECONDS.toMillis(timeOutSec) : 0);
```

Comment:
What happens in case  of timeout ?

Reply:
it throws IOException that will be caught at the very top and gc metadata-iteration fails. so, current gc task fails and next gc task will again run to continue.